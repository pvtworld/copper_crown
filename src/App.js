import React, { Component } from 'react';
import './App.css';
import CopperMap from './Components/CopperMap';

class App extends Component {
  render() {
    return (
    <div className="overlay">
        <CopperMap/>
    </div>
    );
  }
}

export default App;
