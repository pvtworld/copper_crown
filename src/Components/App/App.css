.App {
  text-align: center;
}
.map{
  background-color:#ccc;
  position:fixed;
  width:100%;
  height:100%;
  top:0px;
  left:0px;
  z-index:-1000;
}

.navpage-box {
  background-color: #efefef;
  position: absolute;
  width: 100%;
  height: 100%;
}

/*.modal-body {
    max-height: calc(100vh - 150px);
    overflow-y: auto;
}*/

.center-text{
  text-align: center;
}

.dark-modal{
  margin: 0px;
  padding: 0px;
  background-color: black;
  width: 100%;
  min-height: 100%;
}
.dark-modal > .modal-content{
  font-size: 1.5rem;
  background-color: black;
  color: white;
}

.light-modal{

}
.light-modal > .modal-content{
  font-size: 1.5rem;
  background-color: black;
  color: white;
}

.white-modal{

}

.white-modal > .modal-header{
  background-color: black;
}

.white-modal > .modal-footer{
  background-color: black;
}

.white-modal > .modal-content{
  font-size: 1.5rem;
}

.bg-black {
  background-color: black;
  color: white;
  overflow: hidden;
}

.full-modal{
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
}
.full-modal > .modal-content{
  height: auto;
  min-height: 100%;
  border-radius: 0;
}