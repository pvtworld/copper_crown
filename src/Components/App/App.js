import React from 'react';
import base from '../../Firebase/base';
import LoginContainer from '../LoginContainer/LoginContainer';
import GameContainer from '../GameContainer/GameContainer';
import {getPricePerSquareMeter} from '../../Helpers/PointsHelpers';

export default class App extends React.Component {
    constructor(){
        super();
        this.renderLogin = this.renderLogin.bind(this);
        this.authenticate = this.authenticate.bind(this);
        this.logout = this.logout.bind(this);
        this.authHandler = this.authHandler.bind(this);
        this.addRoof = this.addRoof.bind(this);
        this.roofAlreadyStolen = this.roofAlreadyStolen.bind(this);
        this.getLeader = this.getLeader.bind(this);
    }

    state = {
        userInfo: {
            points: 0,
            areaOfCopper: 0,
        },
        uid: null,
        pricePerSquareMeter: null,
        userLoading: false
    }


        componentWillMount() {
            base.onAuth((user) => {
                if(user) {
                    this.authHandler(null, { user });
                    this.ref = base.syncState(`users/${user.uid}`, {
                        context: this,
                        state: 'userInfo'
                    });
                    this.setState({userLoading: false})
                }
            });
        }
        

    componentDidMount() {
        getPricePerSquareMeter((newPrice) => {
            this.setState({pricePerSquareMeter: newPrice});
        });

    }

    componentWillUnmount() {
        base.removeBinding(this.ref);

    }

    authenticate(provider) {
        console.log(`Trying to log in with ${provider}`); 
        this.setState({userLoading: true})
        base.authWithOAuthPopup(provider, this.authHandler);
    }

    logout() {
        base.unauth();
        console.log('Logging out');
        this.setState({uid: null});
    }

    authHandler(err, authData)  {
        console.log('current user: ');
        console.log(authData)
        if (err) {
            alert(err)
            console.error(err);
            return;
        }
            this.setState({
                uid: authData.user.uid,
                userLoading: false
            });
        };

    getLeader() {
        base.fetch('users', {
            context: this,
            queries: {
                orderByChild: 'points', },
            asArray: true,
            then(response){
                response.reverse().forEach(function(element) {
                    console.log("user: "+element.key + ", points: " +element.points);
                });
            }
        });
    }

    roofAlreadyStolen(newRoof, callback) {
        base.fetch('stolenRoofs', {
            context: this,
            queries: {
                orderByChild: 'roofId',
                equalTo: newRoof.id},
            then(response){
                console.log(Object.keys(response).length)
                callback(newRoof, (Object.keys(response).length));
            }
        });
    }

    addRoof(newPoints, newArea, newRoof) {
        const userInfo= {...this.state.userInfo};
        if(userInfo.points) {
            userInfo.points += parseInt(newPoints, 10);
        } else {
            userInfo.points = parseInt(newPoints, 10);
        }
        if(userInfo.areaOfCopper) {
            userInfo.areaOfCopper += parseInt(newArea, 10);
        } else {
            userInfo.areaOfCopper = parseInt(newArea, 10);
        }
        console.log("adding to firebase")
        base.push('stolenRoofs', {
            data: {roofId: newRoof},
            then(err){
                if(err){
                    console.log(err);
                }
            }
        });

        this.setState({ userInfo });
    }




    renderLogin() {
        return (
            <LoginContainer authenticate={this.authenticate}
                            userLoading={this.state.userLoading}
            />
        )
    }

    render() {
        // check if they are no logged in at all
        if(!this.state.uid) {
            return <div>{this.renderLogin()}</div>
        }

        return (
            <div>
                <GameContainer state={this.state}
                               addRoof={this.addRoof}
                               roofAlreadyStolen={this.roofAlreadyStolen}
                               logout={this.logout}
                               getLeader={this.getLeader}
                               isLoading={this.state.userLoading}
                />
            </div>
        )
    }
}