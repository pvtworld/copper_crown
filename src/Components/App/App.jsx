import React from 'react'
import {Route, Redirect, Switch, BrowserRouter} from 'react-router-dom';
import CopperMap from '../CopperMap/CopperMap';
import Login from '../Login';
import firebase from '../../Firebase/firebase';
import Welcome from '../Welcome';

function PublicRoute ({component: Component, authed, ...rest}) {
    return (
        <Route {...rest} render={(props) => authed === false
            ? <Component {...props} />
            : <Redirect to='/coppermap' />}
        />
    )
}

function PrivateRoute ({component: Component, authed, ...rest}) {
    return (
        <Route
            {...rest} render={(props) => authed === true
                ? <Component {...props} />
                : <Redirect to={{pathname: '/', state: {from: props.location}}} />}
        />
    )
}

export default class App extends React.Component {
    constructor(){
        super();
        this.state = {
            authed: false,
            points: 0
        }
    }
    componentDidMount () {


    componentWillMount() {
        this.removeListener = firebase.auth().onAuthStateChanged((user) => {
            if (user) {
                this.setState({
                    authed: true
                });

                this.ref = base.syncState(`users/${user.uid}`, {
                    context: this,
                    state: 'userInfo'
                });

                console.log('User IS auth');
            } else {
                this.setState({
                    authed: false
                });
                console.log('User NOT auth');
            }
        })
    }
    componentWillUnmount () {
        this.removeListener()
    }
    render() {
        return (
            <BrowserRouter>
                <Switch>
                    <Route path='/' exact component={Welcome} />
                    <PublicRoute authed={this.state.authed} path='/login' component={Login} />
                    <PrivateRoute authed={this.state.authed} path='/coppermap' component={CopperMap} />
                    <Route render={() => <h3>404 feterror</h3>} />
                </Switch>
            </BrowserRouter>
        );
    }
}