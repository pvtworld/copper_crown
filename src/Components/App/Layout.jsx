/*import React from 'react'
import Nav from '../Navigation/Navigationbar'

var Layout = (props) => (
    <div>
        <Nav/>
        {props.children}
    </div>
)

export default Layout;*/
