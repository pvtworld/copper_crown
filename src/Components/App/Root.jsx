import React, { PropTypes } from 'react';
import { Provider } from 'react-redux';
import router from './Router'

const Root = ({ store }) => (
    <Provider store={store}>
        {router}
    </Provider>
);

Root.propTypes = {
    store: PropTypes.object.isRequired,
};

export default Root;