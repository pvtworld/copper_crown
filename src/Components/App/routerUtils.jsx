import { UserAuthWrapper } from 'redux-auth-wrapper'
import { browserHistory } from 'react-router'
import { pathToJS } from 'react-redux-firebase'

export const UserIsAuthenticated = UserAuthWrapper({
    wrapperDisplayName: 'UserIsAuthenticated',
    authSelector: ({ firebase }) => pathToJS(firebase, 'auth'),
    authenticatingSelector: ({ firebase }) =>
    (pathToJS(firebase, 'auth') === undefined) ||
    (pathToJS(firebase, 'isInitializing') === true),
    predicate: auth => auth !== null,
    redirectAction: newLoc => (dispatch) => {
        browserHistory.replace(newLoc);
        dispatch({
            type: 'UNAUTHED_REDIRECT',
            payload: { message: 'User is not authenticated.' }
        })
    }
})

export const UserIsNotAuth = UserAuthWrapper({
    wrapperDisplayName: 'UserIsNotAuthenticated',
    allowRedirectBack: false,
    failureRedirectPath: (state, props) =>
        // redirect to page user was on or to list path
    props.location.query.redirect || '/app',
    authSelector: ({ firebase }) => pathToJS(firebase, 'auth'),
    authenticatingSelector: ({ firebase }) =>
    (pathToJS(firebase, 'auth') === undefined) ||
    (pathToJS(firebase, 'isInitializing') === true),
    predicate: auth => auth === null,
    redirectAction: newLoc => (dispatch) => {
        browserHistory.replace(newLoc);
        dispatch({ type: 'AUTHED_REDIRECT' })
    }
})

export default { UserIsAuthenticated, UserIsNotAuth }