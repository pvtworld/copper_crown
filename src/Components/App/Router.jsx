import React from 'react';
import { Router, Route, browserHistory, IndexRoute } from 'react-router';
import NotFound from '../NotFound/NotFound';
import LoginContainer from '../LoginContainer/LoginContainer';
import CopperMap from '../CopperMap/CopperMap';
import App from './App';

export default (
    <Router history={browserHistory}>
        <Route path="/">
            <Route path="/app" component={App} />
            <Route path="/profile" component={() => <h1>Profile</h1>} />
            <Route path="/leaderboard" component={() => <h1>Leaderboard</h1>} />
            <Route path="/about" component={() => <h1>About</h1>} />
            <IndexRoute component={LoginContainer} />
        </Route>
    <Route path="*" component={NotFound} />
</Router>
);
