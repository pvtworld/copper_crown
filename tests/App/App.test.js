import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../../src/Components/App/App';


beforeAll(() => {
    // Clears the database and adds some testing data.
    // Jest will wait for this promise to resolve before running tests.
    return stubGoogleAPIS();
});

describe('Rendertests', () => {

    it('renders loginScreen when user not logged in', () => {
        const component = renderer.create(<App />);

        expect(component).toMatchSnapshot();
    })

    it('renders map when user logged in', () => {
        const component = shallow(<App/>);
        component.setState({authed : true});
        component.setState({uid : 1});
        component.setState({pricePerSquareMeter : null});

        console.log(component.state('authed'));
        console.log(component.state('uid'));
        console.log(component.state('pricePerSquareMeter'));
        
        expect(toJson(component)).toMatchSnapshot();
    })
})










// Stubbing the google Api below.

window.stubGoogleAPIS = function () {
    return window.google = {
        maps: {
            Animation: {},
            BicyclingLayer: function () {
            },
            Circle: function () {
            },
            ControlPosition: {},
            Data: function () {
            },
            DirectionsRenderer: function () {
            },
            DirectionsService: function () {
            },
            DirectionsStatus: {},
            DirectionsTravelMode: {},
            DirectionsUnitSystem: {},
            DistanceMatrixElementStatus: {},
            DistanceMatrixService: function () {
            },
            DistanceMatrixStatus: {},
            ElevationService: function () {
            },
            ElevationStatus: {},
            FusionTablesLayer: function () {
            },
            Geocoder: function () {
            },
            GeocoderLocationType: {},
            GeocoderStatus: {},
            GroundOverlay: function () {
            },
            ImageMapType: function () {
            },
            InfoWindow: function () {
            },
            KmlLayer: function () {
            },
            KmlLayerStatus: {},
            LatLng: function () {
            },
            LatLngBounds: function () {
            },
            MVCArray: function () {
            },
            MVCObject: function () {
            },
            Map: function () {
                return {
                    setTilt: function () {
                    },
                    mapTypes: {
                        set: function () {
                        }
                    },

                    setOptions:  function () {},
                    getZoom:  function () {},

                    overlayMapTypes: {
                        insertAt: function () {
                        },
                        removeAt: function () {
                        }
                    }
                };
            },
            MapTypeControlStyle: {},
            MapTypeId: {
                HYBRID: '',
                ROADMAP: '',
                SATELLITE: '',
                TERRAIN: ''
            },
            MapTypeRegistry: function () {
            },
            Marker: function () {
                return {
                    setMap: function () {

                    }
                }
            },
            MarkerImage: function () {
            },
            MaxZoomService: function () {
                return {
                    getMaxZoomAtLatLng: function () {
                    }
                };
            },
            MaxZoomStatus: {},
            NavigationControlStyle: {},
            OverlayView: function () {
            },
            Point: function () {
            },
            Polygon: function () {
            },
            Polyline: function () {
            },
            Rectangle: function () {
            },
            SaveWidget: function () {
            },
            ScaleControlStyle: {},
            Size: function () {
            },
            StreetViewCoverageLayer: function () {
            },
            StreetViewPanorama: function () {
            },
            StreetViewService: function () {
            },
            StreetViewStatus: {},
            StrokePosition: {},
            StyledMapType: function () {
            },
            SymbolPath: {},
            TrafficLayer: function () {
            },
            TransitLayer: function () {
            },
            TransitMode: {},
            TransitRoutePreference: {},
            TravelMode: {},
            UnitSystem: {},
            ZoomControlStyle: {},
            __gjsload__: function () {
            },
            event: {
                addListener: function () {
                }
            },
            places: {
                AutocompleteService: function () {
                    return {
                        getPlacePredictions: function () {
                        }
                    };
                }
            }
        }
    };
};


