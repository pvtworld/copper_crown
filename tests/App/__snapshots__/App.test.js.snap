exports[`Rendertests renders loginScreen when user not logged in 1`] = `
<div>
  <div>
    <div
      id="center_text">
      <h2>
        COPPER
      </h2>
      <h3>
        CROWN
      </h3>
    </div>
    <div
      className="col-md-4 col-md-offset-4">
      <div
        className="login panel panel-default"
        id={undefined}>
        <div
          className="panel-heading">
          <h2
            className="panel-title">
            Sign in to play the game
          </h2>
        </div>
        <div
          className="panel-body">
          <button
            className="btn btn-lg btn-primary btn-block"
            disabled={false}
            onClick={[Function]}
            type="button">
            Log In with Facebook
          </button>
          <button
            className="btn btn-lg btn-default btn-block"
            disabled={false}
            onClick={[Function]}
            type="button">
            Log In with Github
          </button>
          <button
            className="btn btn-lg btn-danger btn-block"
            disabled={false}
            onClick={[Function]}
            type="button">
            Log In with Google
          </button>
          <div
            className="text-right ">
            <a
              href={undefined}
              onBlur={[Function]}
              onClick={null}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}>
              Help
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Rendertests renders map when user logged in 1`] = `
<div>
  <GameContainer
    addRoof={[Function]}
    getLeader={[Function]}
    logout={[Function]}
    roofAlreadyStolen={[Function]}
    state={
      Object {
        "authed": true,
        "uid": 1,
        "userInfo": Object {
          "areaOfCopper": 0,
          "points": 0,
        },
      }
    } />
</div>
`;
